### PR TITLE
MON-3896: Add OpenTelemetry operator metrics

### PR DESCRIPTION
**Pull request by rubenvp8510**
Here is the PR on upstream that define the metrics: https://github.com/open-telemetry/opentelemetry-operator/pull/2825/files#diff-c70020b8d6a709d7d5ffaa8c001fa493aac0dfbda90ad9dd0975768e9cfa9417

We also created recording rules on midstream.

* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by rubenvp8510** [comment]
@simonpasquier This one is for opentelemetry.

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -846,6 +846,36 @@ data:
...
+    - '{__name__="type:opentelemetry_collector_extensions:sum",type="zpagesextension|ballastextension|memorylimiterextension|jaegerremotesampling|healthcheckextension|pprofextension|oauth2clientauthextension|oidcauthextension|bearertokenauthextension|filestorage"}'
+    #
+    # owners: (@tracing-team)
+    #
+    # Number of OpenTelemetry collectors used certain connector type
+    - '{__name__="type:opentelemetry_collector_connectors:sum",type="spanmetricsconnector|forwardconnector"}'
+    #
+    # owners: (@tracing-team)
+    #
+    # Number of OpenTelemetry collectors deployed usign certain deployment type
```
Comment (juzhao) [inline Documentation/data-collection.md:876]:
```suggestion
    # Number of OpenTelemetry collectors deployed using certain deployment type
```

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 2]
```diff
@@ -838,6 +838,36 @@ data:
     # Number of Tempo stacks with Jaeger UI enabled/disabled.
     - '{__name__="enabled:tempo_operator_tempostack_jaeger_ui:sum",enabled="true|false"}'
     #
+    # owners: (@tracing-team)
+    #
+    # Number of OpenTelemetry collectors used certain receiver type
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:843]: (nit)  same for all other comments.
```suggestion
    # Number of OpenTelemetry collectors using certain receiver types.
```

**Review by simonpasquier — COMMENTED** [review]
I've got minor remarks as well as @juzhao but virtual lgtm for me.

**Review by simonpasquier — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**